### PR TITLE
[LCN] Fixes problems with fingerprint codes which were received in decimal

### PR DESCRIPTION
**Pull request by andrexp**
fingerprint codes seem also be sent in decimal (as remote control codes) by PC(H)K. The current version of the binding only accepts .ZFmmnnoo codes. This fix supports both .ZFmmnnoo or .ZFmmmnnnooo in decimal.

### CODE REVIEW

File: bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleCodeSubHandler.java [hunk 1]
```diff
@@ -51,7 +53,7 @@ public void handleRefresh(LcnChannelGroup channelGroup, int number) {
     public void handleStatusMessage(Matcher matcher) {
         String code;
 
-        if (matcher.pattern() == FINGERPRINT_PATTERN) {
+        if (matcher.pattern() == FINGERPRINT_PATTERN_HEX) {
```
Comment (fwolter) [inline bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleCodeSubHandler.java:56]: Is there a reason not checking the decimal representation anymore?

**Review by fwolter — CHANGES_REQUESTED** [review]
Thanks for your contribution! Could you add a test case checking the new pattern? https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.lcn/src/test/java/org/openhab/binding/lcn/internal/subhandler

File: bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleCodeSubHandler.java (hunk 1 above)
Comment (andrexp) [reply bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleCodeSubHandler.java:56]: IMHO is the "if" case is only for handling the integer parsing. It is only necessary for the hexadecimal representation to handle the regex result differently:

`Integer.parseInt(matcher.group("byte0"), 16)`

**Comment by andrexp** [comment]
> Thanks for your contribution! Could you add a test case checking the new pattern? https://github.com/openhab/openhab-addons/tree/main/bundles/org.openhab.binding.lcn/src/test/java/org/openhab/binding/lcn/internal/subhandler

I can't find the right file for the test cases for the "code" subhandler. Where are the existing test cases for the other (old fingerprint) code patterns?

File: bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleCodeSubHandler.java (hunk 1 above)
Comment (fwolter) [reply bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleCodeSubHandler.java:56]: You're right! The original code could be refactored to be much simpler, though...

**Comment by fwolter** [comment]
I'm afraid there is none. I don't know why it is missing as all other subhandler do have tests... Maybe I omitted them because I don't have access to a fingerprint sensor and couldn't generate these PCK messages. Can you add a new test file? You can raise your hand if you need any help!

**Comment by andrexp** [comment]
Done, but I have only added the required test for my edits on the code-subhandler. May someone should add the left-over code types as from UT and RT?? I think for this PR this should be not necessary.

**Review by fwolter — APPROVED** [review]
Thank you, LGTM!

**Pull request merged**